### PR TITLE
[SIW-291] Add PID module

### PR DESCRIPTION
**Pull request by balanza**
#### List of Changes

* add `PID` module

#### Motivation and Context

Decode and verify PID on react native

#### Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.

### CODE REVIEW

File: src/pid/sd-jwt/index.ts [hunk 1]
```diff
@@ -0,0 +1,108 @@
+import base64decode from "../lib/base64";
```
Comment (grausof) [inline src/pid/sd-jwt/index.ts:1]: Where is `lib` folder?

File: src/pid/sd-jwt/index.ts [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+  // token are expected in the form "sd-jwt~disclosure0~disclosure1~...~disclosureN"
+  const [rawSdJwt = "", ...rawDisclosures] = token.split("~");
+
+  // get the sd-jwt as object
+  // validate it's a valid SD-JWT for Verifiable Credentials
+  const [header, payload] = rawSdJwt
+    .split(".")
+    .slice(0, 2)
+    .map(base64decode)
+    .map((e) => JSON.parse(e));
```
Comment (grausof) [inline src/pid/sd-jwt/index.ts:30]: ok for now. This part must then be replaced with https://github.com/pagopa/io-react-native-jwt

Comment (grausof) [inline src/pid/sd-jwt/index.ts:30]: json parse may fail!

**Review by grausof — APPROVED** [review]
LGTM. Some things to fix as reported by the code review pipeline

File: src/pid/sd-jwt/index.ts (hunk 1 above)
Comment (hevelius) [reply src/pid/sd-jwt/index.ts:1]: To avoid redefine some methods could base64encode/decode be imported from `io-react-native-jwt`?

Comment (grausof) [reply src/pid/sd-jwt/index.ts]: done in #7

File: src/pid/sd-jwt/index.ts (hunk 2 above)
Comment (grausof) [reply src/pid/sd-jwt/index.ts]: done in #7

Comment (grausof) [reply src/pid/sd-jwt/index.ts]: done in #7

**Comment by grausof** [comment]
Added new features #7

**Pull request merged**